app/index.js

```diff
@@ -54,7 +54,8 @@ module.exports = generators.Base.extend({
         projectRoot: 'generators',
         skipInstall: this.options.skipInstall,
         readme: readmeTpl({
-          generatorName: this.props.name
+          generatorName: this.props.name,
+          generatorRealName: this.props.name.replace('generator-', '')
```

Comment:
Say the name in package json is `generator-test`, when the project is generated the readme should have the name `yo test` not `yo generator-test`. 
how about `yoName` instead of `generatorRealName`? 

Rather than adding application logic on a template this way it is much clear I guess.